base_custom_attribute/models/attribute_attribute.py

```diff
@@ -36,75 +35,83 @@ class AttributeAttribute(models.Model):
-            parent = etree.SubElement(parent, "group", colspan="2", col="4")
+        if self.ttype in ["many2many", "text"]:
+            # Add a separator, sibling of field
             etree.SubElement(
                 parent,
                 "separator",
-                string="%s" % attribute.field_description,
-                colspan="4",
+                string="%s" % self.field_description,
+                colspan="2",
             )
             kwargs["nolabel"] = "1"
-            if attribute.ttype == "many2many":
+            if self.ttype == "many2many":
                 # TODO use an attribute field instead
                 # to let user specify the widget. For now it fixes:
                 # https://github.com/shopinvader/odoo-pim/issues/2
                 kwargs["widget"] = "many2many_tags"
 
-        if attribute.ttype in ["many2one", "many2many"]:
-            if attribute.relation_model_id:
+        if self.ttype in ["many2one", "many2many"]:
+            if self.relation_model_id:
                 # attribute.domain is a string, it may be an empty list
                 try:
-                    domain = ast.literal_eval(attribute.domain)
+                    domain = ast.literal_eval(self.domain)
                 except ValueError:
                     domain = None
                 if domain:
-                    kwargs["domain"] = attribute.domain
+                    kwargs["domain"] = self.domain
                 else:
-                    ids = [op.value_ref.id for op in attribute.option_ids]
+                    ids = [op.value_ref.id for op in self.option_ids]
                     kwargs["domain"] = "[('id', 'in', %s)]" % ids
+                # Add color options if the attribute's Relational Model
+                # has a color field
+                relation_model_obj = self.env[self.relation_model_id.model]
+                if 'color' in relation_model_obj.fields_get().keys():
+                    kwargs["options"] = "{'color_field': 'color'}"
             else:
-                kwargs["domain"] = "[('attribute_id', '=', %s)]" % (
-                    attribute.attribute_id.id
-                )
+                kwargs["domain"] = "[('attribute_id', '=', %s)]" % (self.id)
 
-        kwargs["context"] = "{'default_attribute_id': %s}" % (
-            attribute.attribute_id.id
-        )
-
-        kwargs["required"] = str(
-            attribute.required or attribute.required_on_views
-        )
+        kwargs["context"] = "{'default_attribute_id': %s}" % (self.id)
+        kwargs["required"] = str(self.required or self.required_on_views)
 
         field = etree.SubElement(parent, "field", **kwargs)
-        setup_modifiers(field, self.fields_get(attribute.name))
+        setup_modifiers(field, self.fields_get(self.name))
 
-        return parent
+        return field
 
     @api.model
-    def _build_attributes_notebook(self, attribute_group_ids):
+    def _build_attributes_notebook(self, attribute_ids):
```

Comment:
After thinking twice maybe it will be better to drop this notebook.
Most of customer to not have a lot of attributes
Maybe it's better to simply build group (with a string) per attribute group.
So instead of having the attribute in multiple page, we have one page but with several separator

Technique
field_1                             field_2
field_3                             field_4

Description
field_5                            field_6

....